File: src/Client_Python/yuen_client/Controller/subpages/HowToPlayController.py
'''from client.controller import ClientApplicationController
from client.model.subpages import HowToPlayModel
from client.view.subpages import HowToPlayView
from shared import SwingResources

Class HowToPlayController:
    def __init__(self, view, model, parent):
        self.view = view
        self.model = model
        self.parent = parent

        # action listeners
        self.view.get_btn_join_game.bind("<Button-1>", lambda event: HomeController.JoinGameListener(parent))
        self.view.get_btn_back.bind("<Button-1>", lambda event: parent.view.show_home())

        # mouse listeners
        self.view.get_btn_join_game.bind("<Enter>", lambda event: SwingResources.CursorChanger(self.view.btn_join_game))
        self.view.get_btn_back.bind("<Enter>", lambda event: SwingResources.CursorChanger(self.view.btn_back))
'''

File: src/Client_Java/controller/subpages/HowToPlayController.java
package Client_Java.controller.subpages;

import Client_Java.controller.ClientApplicationController;
import Client_Java.model.subpages.HowToPlayModel;
import Client_Java.view.subpages.HowToPlayView;
import shared.SwingResources;

/**
 * The HowToPlayController processes going back to the home page and joining a game.
 */
public class HowToPlayController {
    /**
     * The view.
     */
    private HowToPlayView view;
    /**
     * The model.
     */
    private HowToPlayModel model;
    /**
     * The parent controller.
     */
    private ClientApplicationController parent;

    /**
     * Constructs a HowToPlayController with a specified view and model.
     * @param view The specified view.
     * @param model The specified model.
     */
    public HowToPlayController(HowToPlayView view, HowToPlayModel model, ClientApplicationController parent) {
        this.view = view;
        this.model = model;
        this.parent = parent;

        // action listeners
        view.getBtnJoinGame().addActionListener(new HomeController.JoinGameListener(parent));
        view.getBtnBack().addActionListener(e -> parent.getView().showHome());

        // mouse listeners
        view.getBtnJoinGame().addMouseListener(new SwingResources.CursorChanger(view.getBtnJoinGame()));
        view.getBtnBack().addMouseListener(new SwingResources.CursorChanger(view.getBtnBack()));
    }
}
